lib/fluent/plugin/out_forward.rb

```diff
@@ -460,9 +460,8 @@ def resolved_host
       end
 
       def resolve_dns!
-        @sockaddr = Socket.pack_sockaddr_in(@port, @host)
-        port, resolved_host = Socket.unpack_sockaddr_in(@sockaddr)
-        return resolved_host
+        # shuffle to support dns round robin
+        Resolv.getaddresses(@host).shuffle!.first
```

Comment:
How about using `Array#sample` like `Resolv.getaddresses(@host).sample`? It doesn't generate a temporary `Array`.